x/stake/keeper/slash.go

```diff
@@ -0,0 +1,57 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	crypto "github.com/tendermint/go-crypto"
+)
+
+// slash a validator
+func (k Keeper) Slash(ctx sdk.Context, pubkey crypto.PubKey, height int64, fraction sdk.Rat) {
```

Comment:
(note that these functions are no longer accurate, they need to deal with the bonding period)

Reply:
k - added a comment